### PR TITLE
Refactor client to use react router

### PR DESCRIPTION
**Pull request by dting**
Stop interval from triggering db queries.

TODO: Broadcast added messages.

### CODE REVIEW

File: client/src/containers/Home/Home.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    });
+  }
+
+  componentWillMount() {
+    if (!this.state.location) {
+      browserHistory.push('/roaming');
+    }
+    this.setState({ username: localStorage.getItem('username') });
+    this.updateLocationState();
+    this.interval = setInterval(this.updateLocationState.bind(this), 500);
```
Comment (wayneadams) [inline client/src/containers/Home/Home.js]: Is this better than using [navigator.geolocation.watchposition](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/watchPosition)?

Comment (dting) [inline client/src/containers/Home/Home.js]: done.

**Comment by wayneadams** [comment]
solid. lgtm!

File: client/src/components/chat/room/MessageList.js [hunk 2]
```diff
@@ -5,8 +5,8 @@ import { MessageListEntry } from './MessageListEntry';
 export const MessageList = (props) => (
   <Panel style={{ fontWeight: 'bold' }} header="Chatroom messages">
     <ListGroup fill>
-      {props.messages.map((message) => (
-        <MessageListEntry message={message} />)
+      {props.room && props.room.messages.slice().reverse().map((message) => (
```
Comment (hannahhenderson) [inline client/src/components/chat/room/MessageList.js]: Change order to messages.map(()=>()).reverse() to avoid using slice.

Comment (dting) [inline client/src/components/chat/room/MessageList.js]: Done.

**Comment by hannahhenderson** [comment]
LGTM

**Pull request merged**